src/atomicvar.h

```diff
@@ -103,6 +114,13 @@
 } while(0)
 #define atomicSetWithSync(var,value) \
     atomic_store_explicit(&var,value,memory_order_seq_cst)
+/* Atomic flag definitions */
+#define redisAtomicFlag atomic_flag
+#define REDIS_ATOMIC_FLAG_INIT ATOMIC_FLAG_INIT
+#define atomicFlagTestSet(var, testres_var) \
+    testres_var = atomic_flag_test_and_set_explicit(&var,memory_order_relaxed)
+#define atomicFlagClear(var) \
+    atomic_flag_clear_explicit(&var,memory_order_relaxed)
 #define REDIS_ATOMIC_API "c11-builtin"
 
 #elif !defined(__ATOMIC_VAR_FORCE_SYNC_MACROS) && \
```

Comment:
@ShooterIT 
Hey :)
Can you please explain what's the purpose of this section and in what scenario it will be activated?

According to [GCC](https://gcc.gnu.org/onlinedocs/gcc/_005f_005fatomic-Builtins.html)
> Note that the ‘__atomic’ builtins assume that programs will conform to the C++11 memory model.

I do see that your intention ([#7707](https://github.com/redis/redis/pull/7707)) was to support earlier versions of GCC, where `stdatomic.h` is missing, and we can't use `_Atomic` types. In this case, we compile with std99 which, as far as I understand, __atomic* functions family are also not included in, so we anyway need to choose the __sync* family.

Reply:
@meiravgri sorry for delayed reply
AFAIK, before gcc11, we just use `_atomic*` functions, it works well when compiling with std99. and when we use old gcc or old redis (before 6.0), i ever found `atomicvar_api:atomic-builtin` in `info` command output. so this section makes sense